Use core compression

kdt3rd opened the pull request:
Switches the C++ compression classes to use the core compression

kdt3rd committed:
Add a temporary context

This will allow us to have a context without having to actually have a
file (i.e. to use a header object in the C++ layer, or compression)

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
[remove] Pull fixes from mainline

This will conflict with main, but is easier to test exact matches with
getting the compression options inline with both libraries

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Expose compression / decompression pipeline function

This will allow users of the pipeline to run the compression /
decompression, and to replace the C++ compressor utilities. This also
adds some additional checks and looser memory restrictions to the
decompression scheme for ease when we don't have all the information
needed for encoding / decoding.

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Add a tag to the constructors to indicate read / write / temp

This makes construction consistent with core, where by once constructed,
the object is ready to be used. (file is opened and parsed or ready for
population)

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
switch compression to use core encode / decode pipeline function

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Ensure common type for std::max

Not all compilers coerce the types to the same

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
remove unneeded vararg from macro

not all compilers have named vararg macros yet

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
remove friend cheat as windows symbols change

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
avoid potential UB math, ensure value is initialized

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd committed:
Avoid oss fuzz checks in c++ by initializing scans differently

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

Vertexwahn commented:
On the Bazel side I get a several of those errors here:

```
modified.exr: (EXR_ERR_BAD_CHUNK_LEADER) Invalid part number reconstructing chunk table: expect 0, found 1
modified.exr: (EXR_ERR_BAD_CHUNK_LEADER) Invalid part number reconstructing chunk table: expect 0, found 1
modified.exr: (EXR_ERR_BAD_CHUNK_LEADER) Invalid part number reconstructing chunk table: expect 0, found 1
modified.exr: (EXR_ERR_BAD_CHUNK_LEADER) Invalid part number reconstructing chunk table: expect 0, found 1
```

`bazel test --test_output=errors //src/examples:OpenEXRExamples`

kdt3rd committed:
Remove over-optimization

Previous commits had tried to reduce the number of calls to chunk offset
but that prevented updates during partial header changes

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>

kdt3rd commented:
> On the Bazel side I get a several of those errors here:
> 
> ```
> modified.exr: (EXR_ERR_BAD_CHUNK_LEADER) Invalid part number reconstructing chunk table: expect 0, found 1
> modified.exr: (EXR_ERR_BAD_CHUNK_LEADER) Invalid part number reconstructing chunk table: expect 0, found 1
> modified.exr: (EXR_ERR_BAD_CHUNK_LEADER) Invalid part number reconstructing chunk table: expect 0, found 1
> modified.exr: (EXR_ERR_BAD_CHUNK_LEADER) Invalid part number reconstructing chunk table: expect 0, found 1
> ```
> 
> `bazel test --test_output=errors //src/examples:OpenEXRExamples`

yeah, the same thing was happening on the mainline build too, I had been too focused on fixing oss-fuzz failures to re-run the normal tests before pushing

kdt3rd committed:
Move tables, test logic to core now compression is moved

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>
